controllers/storageclassrequest/storageclassrequest_controller_test.go

```diff
@@ -218,6 +267,201 @@ func TestProfileReconcile(t *testing.T) {
 	assert.NoError(t, err, caseLabel)
 }
 
+func TestStorageProfileCephBlockPool(t *testing.T) {
+	var err error
+	var caseCounter int
+
+	var primaryTestCases = []struct {
+		label            string
+		expectedPoolName string
+		failureExpected  bool
+		createObjects    []runtime.Object
+		storageProfile   *v1.StorageProfile
+	}{
+		{
+			label:            "valid profile",
+			expectedPoolName: "test-valid-blockpool",
+			failureExpected:  false,
+			storageProfile:   validStorageProfile,
+			createObjects: []runtime.Object{
+				&rookCephv1.CephBlockPool{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-valid-blockpool",
+						Namespace: namespaceName,
+						Labels: map[string]string{
+							controllers.StorageConsumerNameLabel: fakeStorageConsumer.Name,
+							controllers.StorageProfileSpecLabel:  validStorageProfile.GetSpecHash(),
+						},
+					}, Spec: rookCephv1.NamedBlockPoolSpec{
+						Name: "spec",
+						PoolSpec: rookCephv1.PoolSpec{
+							FailureDomain: "zone",
+							DeviceClass:   deviceClass,
+							Parameters:    map[string]string{},
+						},
+					},
+				},
+			},
+		},
+		{
+			label:            "rejected profile",
+			expectedPoolName: "test-rejected-blockpool",
+			failureExpected:  true,
+			storageProfile:   rejectedStorageProfile,
+			createObjects: []runtime.Object{
+				&rookCephv1.CephBlockPool{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-rejected-blockpool",
+						Namespace: namespaceName,
+						Labels: map[string]string{
+							controllers.StorageConsumerNameLabel: fakeStorageConsumer.Name,
+							controllers.StorageProfileSpecLabel:  rejectedStorageProfile.GetSpecHash(),
+						},
+					}, Spec: rookCephv1.NamedBlockPoolSpec{
+						Name: "spec",
+						PoolSpec: rookCephv1.PoolSpec{
+							FailureDomain: "zone",
```

Comment:
zone -> test-zone?

Reply:
I think it's aligned with the naming convention

Reply:
Resolving as per @rchikatw's manual confirmation